README.md

```diff
@@ -1,2 +1,213 @@
+
+    if let edit::Edit::Change(diff) = unity.diff(&not_unity) {
+        assert!(diff[&1].is_copy());
+        assert_eq!(diff[&2].change().unwrap(), &(&2, &3));
+        assert!(diff[&3].is_remove());
+        assert_eq!(diff[&4].insert().unwrap(), &4);
+    } else {
+        unreachable!()
+    }
+}
+```
+
+### Diffus on collections
+Difference between collections is done through the Longest Common Subsequence (LCS) algorithm and additional has support for objects that have changed values but kept its `Same` "identity".
+
+Strings are considered as collections.
+
+The possible edits of a list are `Copy, Insert, Remove, Change`.
+
+```rust
+use diffus_derive::Diffus;
+use diffus::{edit::{self, collection}, Same, Diffable};
+
+#[derive(Diffus, Debug)]
+struct Identified {
+    id: u32,
+    value: u32,
+}
+
+impl Same for Identified {
+    fn same(&self, other: &Self) -> bool {
+        self.id == other.id
+    }
+}
+
+fn main() {
+    let left = vec![
+        Identified { id: 1, value: 0 },
+        Identified { id: 2, value: 0 },
+        Identified { id: 3, value: 0 },
+        Identified { id: 4, value: 0 },
+        Identified { id: 5, value: 0 },
+        Identified { id: 6, value: 0 },
+        Identified { id: 7, value: 0 },
+    ];
+    let right = vec![
+        Identified { id: 1, value: 0 },
+        Identified { id: 2, value: 1 },
+        Identified { id: 4, value: 0 },
+        Identified { id: 3, value: 0 },
+        Identified { id: 5, value: 0 },
+        Identified { id: 6, value: 0 },
+    ];
+
+    let diff = left.diff(&right);
+
+    match diff {
+        edit::Edit::Copy => println!("no difference"),
+        edit::Edit::Change(diff) => {
+            diff.into_iter().map(|edit| {
+                match edit {
+                    collection::Edit::Copy(elem) => println!("copy: {:?}", elem),
+                    collection::Edit::Insert(elem) => println!("insert: {:?}", elem),
+                    collection::Edit::Remove(elem) => println!("remove: {:?}", elem),
+                    collection::Edit::Change(EditedIdentified { id, value}) => {
+                        println!("changed:");
+                        match id {
+                            edit::Edit::Copy => println!("    copy: id"),
+                            edit::Edit::Change((left_id, right_id)) => println!("    id: {} => {}", left_id, right_id),
+                        }
+                        match value {
+                            edit::Edit::Copy => println!("    copy: value"),
+                            edit::Edit::Change((left_value, right_value)) => println!("    value: {} => {}", left_value, right_value),
+                        }
+                    }
+                };
+            }).collect::<Vec<_>>();
+        },
+    };
+}
+```
+
+
+### Diffus on enums
+Difference between two enums works as expected with the addition that it separates changes of the varaint and changes of the inner associated fields.
```

Comment:
separates *between* changes...?
the associated *variant* fields?

"inner" feels a bit confusing